File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/create_model_version_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class CreateModelVersionDetails(object):
    """
    The data to create a ModelVersion.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new CreateModelVersionDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param name:
            The value to assign to the name property of this CreateModelVersionDetails.
        :type name: str

        :param description:
            The value to assign to the description property of this CreateModelVersionDetails.
        :type description: str

        :param source:
            The value to assign to the source property of this CreateModelVersionDetails.
        :type source: str

        :param run_id:
            The value to assign to the run_id property of this CreateModelVersionDetails.
        :type run_id: str

        :param tags:
            The value to assign to the tags property of this CreateModelVersionDetails.
        :type tags: list[oci.aidataplatform_dp.models.ModelVersionTag]

        :param run_link:
            The value to assign to the run_link property of this CreateModelVersionDetails.
        :type run_link: str

        :param model_id:
            The value to assign to the model_id property of this CreateModelVersionDetails.
        :type model_id: str

        """
        self.swagger_types = {
            'name': 'str',
            'description': 'str',
            'source': 'str',
            'run_id': 'str',
            'tags': 'list[ModelVersionTag]',
            'run_link': 'str',
            'model_id': 'str'
        }

        self.attribute_map = {
            'name': 'name',
            'description': 'description',
            'source': 'source',
            'run_id': 'run_id',
            'tags': 'tags',
            'run_link': 'run_link',
            'model_id': 'model_id'
        }

        self._name = None
        self._description = None
        self._source = None
        self._run_id = None
        self._tags = None
        self._run_link = None
        self._model_id = None

    @property
    def name(self):
        """
        **[Required]** Gets the name of this CreateModelVersionDetails.
        Register models under this name.


        :return: The name of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this CreateModelVersionDetails.
        Register models under this name.


        :param name: The name of this CreateModelVersionDetails.
        :type: str
        """
        self._name = name

    @property
    def description(self):
        """
        Gets the description of this CreateModelVersionDetails.
        Description of this model version.


        :return: The description of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this CreateModelVersionDetails.
        Description of this model version.


        :param description: The description of this CreateModelVersionDetails.
        :type: str
        """
        self._description = description

    @property
    def source(self):
        """
        **[Required]** Gets the source of this CreateModelVersionDetails.
        URI indicating the location of the source model artifacts.


        :return: The source of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._source

    @source.setter
    def source(self, source):
        """
        Sets the source of this CreateModelVersionDetails.
        URI indicating the location of the source model artifacts.


        :param source: The source of this CreateModelVersionDetails.
        :type: str
        """
        self._source = source

    @property
    def run_id(self):
        """
        Gets the run_id of this CreateModelVersionDetails.
        Run ID that generated the model version.


        :return: The run_id of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._run_id

    @run_id.setter
    def run_id(self, run_id):
        """
        Sets the run_id of this CreateModelVersionDetails.
        Run ID that generated the model version.


        :param run_id: The run_id of this CreateModelVersionDetails.
        :type: str
        """
        self._run_id = run_id

    @property
    def tags(self):
        """
        Gets the tags of this CreateModelVersionDetails.
        Tags of model version.


        :return: The tags of this CreateModelVersionDetails.
        :rtype: list[oci.aidataplatform_dp.models.ModelVersionTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this CreateModelVersionDetails.
        Tags of model version.


        :param tags: The tags of this CreateModelVersionDetails.
        :type: list[oci.aidataplatform_dp.models.ModelVersionTag]
        """
        self._tags = tags

    @property
    def run_link(self):
        """
        Gets the run_link of this CreateModelVersionDetails.
        Run Link: Direct link to the run that generated this version.


        :return: The run_link of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._run_link

    @run_link.setter
    def run_link(self, run_link):
        """
        Sets the run_link of this CreateModelVersionDetails.
        Run Link: Direct link to the run that generated this version.


        :param run_link: The run_link of this CreateModelVersionDetails.
        :type: str
        """
        self._run_link = run_link

    @property
    def model_id(self):
        """
        Gets the model_id of this CreateModelVersionDetails.
        Model id for model version that is used to link the registered model to the source logged model.


        :return: The model_id of this CreateModelVersionDetails.
        :rtype: str
        """
        return self._model_id

    @model_id.setter
    def model_id(self, model_id):
        """
        Sets the model_id of this CreateModelVersionDetails.
        Model id for model version that is used to link the registered model to the source logged model.


        :param model_id: The model_id of this CreateModelVersionDetails.
        :type: str
        """
        self._model_id = model_id

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/CreateModelVersionDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The data to create a ModelVersion.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=CreateModelVersionDetails.Builder.class)

public final class CreateModelVersionDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"name", "description", "source", "runId", "tags", "runLink", "modelId"})
    public CreateModelVersionDetails(String name, String description, String source, String runId, java.util.List<ModelVersionTag> tags, String runLink, String modelId) {
        super();
        this.name = name;
        this.description = description;
        this.source = source;
        this.runId = runId;
        this.tags = tags;
        this.runLink = runLink;
        this.modelId = modelId;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Register models under this name.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("name")
        private String name;

                /**
         * Register models under this name.
         * @param name the value to set
         * @return this builder
         **/
        

        public Builder name(String name) {
        this.name = name;
        return this;
        }
            /**
     * Description of this model version.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

                /**
         * Description of this model version.
         * @param description the value to set
         * @return this builder
         **/
        

        public Builder description(String description) {
        this.description = description;
        return this;
        }
            /**
     * URI indicating the location of the source model artifacts.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("source")
        private String source;

                /**
         * URI indicating the location of the source model artifacts.
         * @param source the value to set
         * @return this builder
         **/
        

        public Builder source(String source) {
        this.source = source;
        return this;
        }
            /**
     * Run ID that generated the model version.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("run_id")
        private String runId;

                /**
         * Run ID that generated the model version.
         * @param runId the value to set
         * @return this builder
         **/
        

        public Builder runId(String runId) {
        this.runId = runId;
        return this;
        }
            /**
     * Tags of model version.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tags")
        private java.util.List<ModelVersionTag> tags;

                /**
         * Tags of model version.
         * @param tags the value to set
         * @return this builder
         **/
        

        public Builder tags(java.util.List<ModelVersionTag> tags) {
        this.tags = tags;
        return this;
        }
            /**
     * Run Link: Direct link to the run that generated this version.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("run_link")
        private String runLink;

                /**
         * Run Link: Direct link to the run that generated this version.
         * @param runLink the value to set
         * @return this builder
         **/
        

        public Builder runLink(String runLink) {
        this.runLink = runLink;
        return this;
        }
            /**
     * Model id for model version that is used to link the registered model to the source logged model.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("model_id")
        private String modelId;

                /**
         * Model id for model version that is used to link the registered model to the source logged model.
         * @param modelId the value to set
         * @return this builder
         **/
        

        public Builder modelId(String modelId) {
        this.modelId = modelId;
        return this;
        }


        public CreateModelVersionDetails build() {
            CreateModelVersionDetails model = new CreateModelVersionDetails(this.name
                    , this.description
                    , this.source
                    , this.runId
                    , this.tags
                    , this.runLink
                    , this.modelId);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(CreateModelVersionDetails model) {
                this.name(model.getName());
    this.description(model.getDescription());
    this.source(model.getSource());
    this.runId(model.getRunId());
    this.tags(model.getTags());
    this.runLink(model.getRunLink());
    this.modelId(model.getModelId());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Register models under this name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * Register models under this name.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Description of this model version.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Description of this model version.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * URI indicating the location of the source model artifacts.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("source")
    private final String source;

        /**
     * URI indicating the location of the source model artifacts.
     * @return the value
     **/
    
    public String getSource() {
        return source;
    }


        /**
     * Run ID that generated the model version.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("run_id")
    private final String runId;

        /**
     * Run ID that generated the model version.
     * @return the value
     **/
    
    public String getRunId() {
        return runId;
    }


        /**
     * Tags of model version.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<ModelVersionTag> tags;

        /**
     * Tags of model version.
     * @return the value
     **/
    
    public java.util.List<ModelVersionTag> getTags() {
        return tags;
    }


        /**
     * Run Link: Direct link to the run that generated this version.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("run_link")
    private final String runLink;

        /**
     * Run Link: Direct link to the run that generated this version.
     * @return the value
     **/
    
    public String getRunLink() {
        return runLink;
    }


        /**
     * Model id for model version that is used to link the registered model to the source logged model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("model_id")
    private final String modelId;

        /**
     * Model id for model version that is used to link the registered model to the source logged model.
     * @return the value
     **/
    
    public String getModelId() {
        return modelId;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("CreateModelVersionDetails(");
        sb.append("name=").append(String.valueOf(this.name));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", source=").append(String.valueOf(this.source));
        sb.append(", runId=").append(String.valueOf(this.runId));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(", runLink=").append(String.valueOf(this.runLink));
        sb.append(", modelId=").append(String.valueOf(this.modelId));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CreateModelVersionDetails)) {
            return false;
        }

        CreateModelVersionDetails other = (CreateModelVersionDetails) o;
        return java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.source, other.source) &&
            java.util.Objects.equals(this.runId, other.runId) &&
            java.util.Objects.equals(this.tags, other.tags) &&
            java.util.Objects.equals(this.runLink, other.runLink) &&
            java.util.Objects.equals(this.modelId, other.modelId);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.source == null ? 43 : this.source.hashCode());
        result = (result * PRIME) + (this.runId == null ? 43 : this.runId.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        result = (result * PRIME) + (this.runLink == null ? 43 : this.runLink.hashCode());
        result = (result * PRIME) + (this.modelId == null ? 43 : this.modelId.hashCode());
        return result;
    }


}
